samples/match/match_enum_else_binding.jakt

```diff
@@ -0,0 +1,24 @@
+/// Expect:
+/// - output: "I found an A 1\nJust 2\n"
```

Comment:
This should be "Just 1" rather than "I found an A 1"

Reply:
Do you mean because if the else is first it should override the A match?
The current behavior without else bindings is:
```
let a = Foo::A

match a {
    else => println("else")
    A => println("I found an A")
}
```
Will print `I found an A`

Reply:
That looks like it's a bug to me, since there isn't a way for `else` to be a catchall if your cases overlap in some way.

Pinging @alimpfard for thoughts here too. My assumption was that `match` would work like other languages, where we start matching at the top and go down the list.

Reply:
🤔 I originally made "else" just a name for "all the cases not mentioned in the match" as that was easier to implement, so the order never really mattered (if that part of the IDE extension is still working, hovering else would then show you which cases it covers)

Interestingly, `else` doesn't actually feel like an irrefutable pattern to me, where e.g. `_` does...

Reply:
Both behaviours seem sensible in a way to me, "else" being a name for "everything else" and "else" being an irrefutable pattern, I kinda like the O(1)-ness of the match, but it also isn't consistent with value matches (we do go down the list checking for equality there)

Reply:
@robryanx - if that's how it already works, that shouldn't block this PR, I just wanted to have a clear understanding of where we want to go

Reply:
I'll make an issue for that so we can discuss it there.